catroid/src/main/java/org/catrobat/catroid/utils/HtmlRegexExtractor.java

```diff
@@ -56,87 +57,64 @@ private void showError() {
 				R.string.formula_editor_function_regex_html_extractor_not_found);
 	}
 
-	@VisibleForTesting
-	public String findKeyword(String keyword, String text) {
+	@VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
+	public String findKeyword(String keyword, String html) {
 		if (keyword.equals("")) {
 			return null;
 		}
-		if (text.indexOf(keyword) >= 0) {
+		if (html.contains(keyword)) {
 			return keyword;
 		} else {
-			return findKeywordWithHtmlBetweenWordsInText(keyword, text);
+			return findKeywordWithHtmlBetweenWordsInText(keyword, html);
 		}
 	}
 
-	private String findKeywordWithHtmlBetweenWordsInText(String keyword, String text) {
-		String[] splittedKeyword = keyword.split(" ");
-		String regex = Pattern.quote(splittedKeyword[0]);
-
-		for (int i = 1; i < splittedKeyword.length; i++) {
-			regex += ".*?" + Pattern.quote(splittedKeyword[i]);
-		}
-		return findShortestOccurrenceInText(regex, text);
-	}
+	private String findKeywordWithHtmlBetweenWordsInText(String keywords, String html) {
 
-	private String findShortestOccurrenceInText(String regex, String text) {
-		Matcher m = Pattern.compile(regex).matcher(text);
+		String[] splittedKeywords = keywords.split("\\s+");
+		String keywordsWithHtmlBetweenWords = Pattern.quote(splittedKeywords[0]);
 
-		String shortestOccurrence = null;
-		int lastIndex = -1;
-		while (m.find(lastIndex + 1)) {
-			String found = m.group();
-			if (shortestOccurrence == null || shortestOccurrence.length() > found.length()) {
-				shortestOccurrence = found;
-				lastIndex = m.start();
-			}
+		for (int i = 1; i < splittedKeywords.length; i++) {
+			keywordsWithHtmlBetweenWords += "(\\s|&nbsp;|<[^>]+>)+?" + Pattern.quote(splittedKeywords[i]);
 		}
-		return shortestOccurrence;
+		Matcher matcher = Pattern.compile(keywordsWithHtmlBetweenWords).matcher(html);
+		if (matcher.find()) {
+			return matcher.group();
+		} 
+		return null;
 	}
 
-	public String htmlToRegexConverter(String keyword, String htmlText) {
-		int keywordIndex;
-		String regex;
+	@VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
+	public String htmlToRegexConverter(String keyword, String html) {
 
 		if (keyword != null) {
-			keywordIndex = htmlText.indexOf(keyword);
-			if (keyword.equals(htmlText)) {
-				regex = "(.*)";
-			} else {
-				regex = "(.*)";
+			int keywordIndex = html.indexOf(keyword);
```

Comment:
PLEASE REFACTOR:
In my opinion, it's a bit hard to read.
My suggestion would be:
```
if (keyword == null) {
	return null;
}
int keywordIndex = html.indexOf(keyword);
if (keyword.equals(html) || keywordIndex < 0) {
	return "(.+)";
}
```

Reply:
> PLEASE REFACTOR:
> ...

Agreed & done, thanks!